File: Household Budget Calculator/[C#]-Household Budget Calculator/C#/Household budget calculator CS/expanding property classes/Household.cs
//using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.ComponentModel;

namespace Household_budget_calculator_CS.expanding_property_classes
{
    [TypeConverter(typeof(ExpandableObjectConverter))]
    [System.Serializable()]
    public class Household
    {
        public event propertiesChangedEventHandler propertiesChanged;
        public delegate void propertiesChangedEventHandler(propertiesChangedEventArgs e);

        private decimal _mortgage;
        [DisplayName("Mortgage/Rent"), RefreshProperties(RefreshProperties.All)]
        public decimal mortgage
        {
            get { return _mortgage; }
            set
            {
                _mortgage = value;
                totalHousehold = mortgage + groceries + clothing + laundry + homehelp;
            }
        }

        private decimal _groceries;
        [DisplayName("Groceries"), RefreshProperties(RefreshProperties.All)]
        public decimal groceries
        {
            get { return _groceries; }
            set
            {
                _groceries = value;
                totalHousehold = mortgage + groceries + clothing + laundry + homehelp;
            }
        }

        private decimal _clothing;
        [DisplayName("Clothing (adult)"), RefreshProperties(RefreshProperties.All)]
        public decimal clothing
        {
            get { return _clothing; }
            set
            {
                _clothing = value;
                totalHousehold = mortgage + groceries + clothing + laundry + homehelp;
            }
        }

        private decimal _laundry;
        [DisplayName("Laundry / Dry Cleaning"), RefreshProperties(RefreshProperties.All)]
        public decimal laundry
        {
            get { return _laundry; }
            set
            {
                _laundry = value;
                totalHousehold = mortgage + groceries + clothing + laundry + homehelp;
            }
        }

        private decimal _homehelp;
        [DisplayName("Home Help"), RefreshProperties(RefreshProperties.All)]
        public decimal homehelp
        {
            get { return _homehelp; }
            set
            {
                _homehelp = value;
                totalHousehold = mortgage + groceries + clothing + laundry + homehelp;
            }
        }

        private decimal _totalHousehold;
        [ReadOnlyAttribute(true), Browsable(false), DisplayName("Total Household Outgoings")]
        public decimal totalHousehold
        {
            get { return _totalHousehold; }
            set
            {
                _totalHousehold = value;
                if (propertiesChanged != null)
                {
                    propertiesChanged(new propertiesChangedEventArgs
                    {
                        propName = "Household",
                        newValue = value
                    });
                }
            }
        }

        public override string ToString()
        {
            return totalHousehold.ToString("c2");
        }


    }
}

File: Household Budget Calculator/[VB]-Household Budget Calculator/VB/Household budget calculator/expanding property classes/Household.vb
Imports System.ComponentModel

<TypeConverter(GetType(ExpandableObjectConverter))> <System.Serializable()> _
Public Class Household

    Public Event propertiesChanged(ByVal e As propertiesChangedEventArgs)

    Private _mortgage As Decimal
    <DisplayName("Mortgage/Rent"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property mortgage() As Decimal
        Get
            Return _mortgage
        End Get
        Set(ByVal value As Decimal)
            _mortgage = value
            totalHousehold = mortgage + groceries + clothing + laundry + homehelp
        End Set
    End Property

    Private _groceries As Decimal
    <DisplayName("Groceries"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property groceries() As Decimal
        Get
            Return _groceries
        End Get
        Set(ByVal value As Decimal)
            _groceries = value
            totalHousehold = mortgage + groceries + clothing + laundry + homehelp
        End Set
    End Property

    Private _clothing As Decimal
    <DisplayName("Clothing (adult)"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property clothing() As Decimal
        Get
            Return _clothing
        End Get
        Set(ByVal value As Decimal)
            _clothing = value
            totalHousehold = mortgage + groceries + clothing + laundry + homehelp
        End Set
    End Property

    Private _laundry As Decimal
    <DisplayName("Laundry / Dry Cleaning"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property laundry() As Decimal
        Get
            Return _laundry
        End Get
        Set(ByVal value As Decimal)
            _laundry = value
            totalHousehold = mortgage + groceries + clothing + laundry + homehelp
        End Set
    End Property

    Private _homehelp As Decimal
    <DisplayName("Home Help"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property homehelp() As Decimal
        Get
            Return _homehelp
        End Get
        Set(ByVal value As Decimal)
            _homehelp = value
            totalHousehold = mortgage + groceries + clothing + laundry + homehelp
        End Set
    End Property

    Private _totalHousehold As Decimal
    <ReadOnlyAttribute(True), _
    Browsable(False), _
    DisplayName("Total Household Outgoings")> _
    Public Property totalHousehold() As Decimal
        Get
            Return _totalHousehold
        End Get
        Set(ByVal value As Decimal)
            _totalHousehold = value
            RaiseEvent propertiesChanged(New propertiesChangedEventArgs With {.propName = "Household", .newValue = value})
        End Set
    End Property

    Public Overrides Function ToString() As String
        Return totalHousehold.ToString("c2")
    End Function

End Class
